File: Data-Agent-Client/src/services/index.service.ts
import http from '../lib/http';
import { ApiPaths } from '../constants/apiPaths';

export interface IndexMetadata {
  name: string;
  type: string;
  columns: string[];
  unique: boolean;
  isPrimaryKey?: boolean;
}

export const indexService = {
  listIndexes: async (
    connectionId: string,
    tableName: string,
    catalog?: string,
    schema?: string
  ): Promise<IndexMetadata[]> => {
    const params: Record<string, string> = { connectionId, tableName };
    if (catalog != null && catalog !== '') params.catalog = catalog;
    if (schema != null && schema !== '') params.schema = schema;

    const response = await http.get<IndexMetadata[]>(ApiPaths.INDEXES, { params });
    return response.data;
  },
};

File: Data-Agent-Server/data-agent-server-app/src/main/java/edu/zsc/ai/domain/service/db/IndexService.java
package edu.zsc.ai.domain.service.db;

import edu.zsc.ai.plugin.model.metadata.IndexMetadata;

import java.util.List;

public interface IndexService {

    List<IndexMetadata> listIndexes(Long connectionId, String catalog, String schema, String tableName, Long userId);
}
